packages/rocketchat-ui/views/app/room.coffee

```diff
@@ -28,7 +28,28 @@ Template.room.helpers
 		return isSubscribed(this._id)
 
 	messagesHistory: ->
-		return ChatMessage.find { rid: this._id, t: { '$ne': 't' }  }, { sort: { ts: 1 } }
+		hideMessagesOfType = []
+		RocketChat.settings.collection.find({_id: /Message_HideType_.+/}).forEach (record) ->
```

Comment:
Is this part of the code being executed every time this method is called? Couldn't we cache this? Maybe on a session variable to be reactive?